### DISCUSSION TITLE
Deployment strategy in docker deployments.

### DISCUSSION THREAD

**Discussion by shivgarg5676** [vercel/next.js#21015] [Help]
Hi, I am running 3 servers in prod (docker deployments on AWS ECS) and all three servers are behind a load balancer. So when I do a deployment,  Builds get updated on one server at a time. Now let say, a deployment is going on and two servers have the latest code and one has the older version of code.  Now, a user having the older build's HTML, requests some chunks and that request gets routed to the server containing the new build (with 66% probability). In this case, his request will fail and experience will deteriorate. How do you manage such downtimes during deployments?

The thing is I don't want to use CDN to cache js as next generates js based on the browser of the user.

**Comment by andrewgadziksonos** [comment]
> next generates js based on the browser of the user

I did not know this!  Where in the docs does it say this?  You could use a CDN and add the useragent header as a cache key

**Comment by stigkj** [comment]
@shivgarg5676 @andrewgadziksonos Next.js is generating 2 different bundles, 1 for modern browsers, i.e. those that handle [`<script type="module">`](https://caniuse.com/es6-module), and 1 for browsers that do not handle it. So it is a little bit misleading to say that Next.js generates js based on the user's browser.

So the bottom line is: use a CDN and older build's HTML won't be a problem.
